cmd/router/main.go

```diff
@@ -277,28 +291,45 @@ func graphHandler(w http.ResponseWriter, req *http.Request) {
 		log.Error(err, "failed to process request")
 		w.Header().Set("Content-Type", "application/json")
 		w.WriteHeader(statusCode)
-		w.Write(prepareErrorResponse(err, "Failed to process request"))
+		if _, err := w.Write(prepareErrorResponse(err, "Failed to process request")); err != nil {
+			log.Error(err, "an error has occurred while writing the response")
+		}
 	} else {
 		if json.Valid(response) {
 			w.Header().Set("Content-Type", "application/json")
 		}
 		w.WriteHeader(statusCode)
-		w.Write(response)
+		if _, err := w.Write(response); err != nil {
+			log.Error(err, "an error has occurred while writing the response")
+		}
+	}
+}
 
+func compilePatterns(patterns []string) []*regexp.Regexp {
+	var compiled []*regexp.Regexp
+	for _, p := range patterns {
+		c, err := regexp.Compile(p)
+		if err != nil {
+			log.Error(err, "failed to compile header pattern", "pattern", p)
```

Comment:
Should we fail explicitly ?

Reply:
Yeah, I feel , we should fail if router fails to come up due to invalid pattern. 
The only downside is that, today this property is controlled only globally via configMap so any typo/mistake in this will not let any of the router come up so blast radius is all IGs in the cluster.

To overcome this, we do have this PR pending https://github.com/kserve/kserve/pull/2710#pullrequestreview-1695636645. Let's try to get that in too. If you can review it , it would be great.

Reply:
@yuzisun Your thoughts ?

Reply:
@rachitchauhan43 Dan feels we should keep it this way as it will affect all the IGs.

Reply:
Ok. We can start with this. At least we are logging so it should be easy to debug.